File: JavaScript solutions/2022. Convert 1D Array Into 2D Array.js
/**
 * @param {number[]} original
 * @param {number} m
 * @param {number} n
 * @return {number[][]}
 */
var construct2DArray = function (original, m, n) {
  const len = original.length;
  const ans = [];
  if (m * n !== len) return ans;
  let v = [];
  let cnt = 0;
  for (let i = 0; i < len; i++) {
    v.push(original[i]);
    cnt++;
    if (cnt % n === 0 && v.length !== 0) {
      ans.push(v);
      v = [];
    }
  }
  return ans;
};

File: C++ solutions/2022. Convert 1D Array Into 2D Array.cpp
class Solution {
public:
    vector<vector<int>> construct2DArray(vector<int>& original, int m, int n) {
        int len=original.size();
        vector<vector<int>>ans;
        if(m*n!=len) return ans;
        vector<int>v,vv;
        int cnt=0;
        for(int i=0;i<len;i++){
            v.push_back(original[i]);
            cnt++;
            if(cnt%n==0 && v.size()!=0){
                ans.push_back(v);
                v=vv;
            }
        } 
        return ans;
    }
};